patroni/ha.py

```diff
@@ -1402,3 +1415,23 @@ def get_remote_member(self, member=None):
 
     def get_remote_master(self):
         return self.get_remote_member()
+
+    def call_pre_promote(self, config):
+        """
+        Runs a fencing script after the leader lock is acquired but before the replica is promoted.
+        If the script exits with a non-zero code, promotion does not happen and the leader key is removed from DCS.
+        """
+        cmd = config.get('pre_promote')
+        if cmd:
+            self._pre_promote_subprocess = CancellableSubprocess()
+            try:
+               ret = self._pre_promote_subprocess.call(shlex.split(cmd))
```

Comment:
It makes sense to add some arguments, like the cluster name (`scope`) and the address and name of the former leader if they are known.

Reply:
i will add/test them once the script itself correctly executes